process/block/baseProcess.go

```diff
@@ -271,7 +271,7 @@ func (bp *baseProcessor) removeLastNotarized() {
 	bp.mutNotarizedHdrs.Lock()
 	for shardId := range bp.notarizedHdrs {
 		notarizedHdrsCount := len(bp.notarizedHdrs[shardId])
-		if notarizedHdrsCount > 0 {
+		if notarizedHdrsCount > 1 {
```

Comment:
I'm against these types of changes without a proper test. Also, notarizedHdrsCount > 1 could have been extracted in a bool var to better understand the case.

Reply:
It was tested, try to launch a node in a public testnet with a wrong config - which results in a different starting genesis block. The problem is we should never revert the first put block. Not a written test. But here was a logical error. Notarized list should never be nil.

Reply:
Added test